anomaly/json_trees_file_loader.cpp

```diff
@@ -0,0 +1,125 @@
+#include "json_trees_file_loader.hpp"
+
+#include <thirdparty/jsonxx/jsonxx_utils.h>
+
+namespace HawkTracer
+{
+namespace anomaly
+{
+
+bool JsonTreesFileLoader::init(const std::string& file_name)
+{
+    _file.open(file_name);
+    if (_file.is_open())
+    {
+        return _parse_file();
+    }
+    return false;
+}
+
+std::vector<std::pair<std::shared_ptr<CallGraph::TreeNode>, int>> JsonTreesFileLoader::get_trees()
+{
+    return _trees;
+}
+
+bool JsonTreesFileLoader::_parse_file()
+{
+    _json_obj.parse(_file);
+    jsonxx::Array trees;
+    try_get_value<jsonxx::Array, jsonxx::Array>(_json_obj, "trees", trees);
+    for (const auto& tree : trees.values())
+    {
+        if (!_parse_tree(*tree))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool JsonTreesFileLoader::_parse_tree(const jsonxx::Value& tree)
+{
+    for (const auto& node : tree.array_value_->values())
+    {
+        if (!_parse_node_data(*node))
+        {
+            return false;
+        }
+    } 
+    for (const auto& node : tree.array_value_->values())
+    {
+        if (!_parse_node_edges(*node))
+        {
+            return false;
+        }
+    }   
+    return true;
+}
+
+bool JsonTreesFileLoader::_parse_node_data(const jsonxx::Value& node)
+{
+    int id;
+    int cnt_calls;
+    int last_start_ts;
+    int last_stop_ts;
+    int total_dur;
+    int total_children_dur;
+    std::string label;
+
+    if (!node.object_value_->has<jsonxx::String>("label") || !node.object_value_->has<jsonxx::Number>("id"))
+    {
+        return false;
+    }
+    try_get_value<std::string, jsonxx::String>(*node.object_value_, "label", label);
+    try_get_value<int, jsonxx::Number>(*node.object_value_, "id", id);
```

Comment:
What if one of those fields don't exist? Shouldn't we fail loading the tree then?